From 68575dd593da700e36a6c91fdc9fc8d088e4b3b2 Mon Sep 17 00:00:00 2001
From: mtk81325 <peng.zhou@mediatek.com>
Date: Mon, 13 Jul 2020 14:56:24 +0800
Subject: [PATCH 460/830] ANDROID: Incremental fs: magic number compatible
 32-bit
X-NVConfidentiality: public

Incfs's magic is bigger than 32-bit, but super block structure's
s_magic is unsigned long which is 32-bit in ARM 32-bit platform.

Do the cast for magic!

Bug: 159772865
Signed-off-by: Peng Zhou <Peng.Zhou@mediatek.com>
Signed-off-by: mtk81325 <peng.zhou@mediatek.com>
Signed-off-by: Paul Lawrence <paullawrence@google.com>
Change-Id: Iae4f38774440c7d6ae44529d4f0f8ebb2ec5dacc
---
 fs/incfs/vfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/incfs/vfs.c b/fs/incfs/vfs.c
index ee581dea17a1..2a48af2431e7 100644
--- a/fs/incfs/vfs.c
+++ b/fs/incfs/vfs.c
@@ -2181,7 +2181,7 @@ struct dentry *incfs_mount_fs(struct file_system_type *type, int flags,
 	sb->s_op = &incfs_super_ops;
 	sb->s_d_op = &incfs_dentry_ops;
 	sb->s_flags |= S_NOATIME;
-	sb->s_magic = INCFS_MAGIC_NUMBER;
+	sb->s_magic = (long)INCFS_MAGIC_NUMBER;
 	sb->s_time_gran = 1;
 	sb->s_blocksize = INCFS_DATA_FILE_BLOCK_SIZE;
 	sb->s_blocksize_bits = blksize_bits(sb->s_blocksize);
-- 
2.17.1

